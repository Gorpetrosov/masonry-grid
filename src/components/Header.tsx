import {memo} from "react";

const Header = ( {headerText} : {headerText: string }) => {
    return (
        <header className="layout-header">
            <h1>{headerText}</h1>
        </header>
    )
}

export default memo(Header);