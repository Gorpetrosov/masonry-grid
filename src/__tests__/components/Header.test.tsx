import {describe, test, expect, vi} from 'vitest';
import { render, screen } from '@testing-library/react';
import Header from "../../components/Header.tsx";

describe('Header Component', () => {

    afterEach(() => {
        vi.resetAllMocks()
    });

    test('real Header component memoization', () => {
        const { rerender } = render(<Header headerText="Test" />);

        const initialHtml = screen.getByRole('heading').outerHTML;

        rerender(<Header headerText="Test" />);

        expect(screen.getByRole('heading').outerHTML).toBe(initialHtml);
    });

    test('matches snapshot', () => {
        const { container } = render(<Header headerText="Test Header" />);
        expect(container).toMatchSnapshot();
    });
});