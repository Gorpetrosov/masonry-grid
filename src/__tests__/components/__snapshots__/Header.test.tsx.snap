// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header Component > matches snapshot 1`] = `
<div>
  <header
    class="layout-header"
  >
    <h1>
      Test Header
    </h1>
  </header>
</div>
`;
